From 2bbb88beb7ca96ebf1d9da611929e816922a8aee Mon Sep 17 00:00:00 2001
From: linmin <linmin@eswincomputing.com>
Date: Mon, 27 May 2024 14:07:34 +0800
Subject: [PATCH 059/109] WIN2030-15279:fix(console):Print temperature in
 Celcius format

Changelogs:
1.Make temp command print temperature in Celcius format

Change-Id: I20b4b8a09941a82aed9d2ef8c30b8cd5ff79b38e
---
 Core/Src/console.c             | 10 ++++++----
 Core/Src/hf_protocol_process.c |  2 +-
 2 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/Core/Src/console.c b/Core/Src/console.c
index 7bd7a84..0d8ef3d 100644
--- a/Core/Src/console.c
+++ b/Core/Src/console.c
@@ -887,7 +887,7 @@ static BaseType_t prvCommandRtcGet(char *pcWriteBuffer, size_t xWriteBufferLen,
         default:
             break;
     }
-    snprintf(pcWriteBuffer, xWriteBufferLen, "%u-%u-%u %s %u:%u:%u CST\r\n",
+    snprintf(pcWriteBuffer, xWriteBufferLen, "%u-%02u-%2u %s %02u:%02u:%02u CST\r\n",
                 date.Year, date.Month, date.Date, cWeekDay,
                 time.Hours, time.Minutes, time.Seconds);
 
@@ -978,8 +978,10 @@ static BaseType_t prvCommandTempGet(char *pcWriteBuffer, size_t xWriteBufferLen,
          snprintf(pcWriteBuffer, xWriteBufferLen, "Faild to get PVT info(errcode:%d)\n", ret);
     }
     else {
-        snprintf(pcWriteBuffer, xWriteBufferLen,"cpu_temp:%d  npu_temp:%d  fan_speed:%d\n",
-            pvtInfo.cpu_temp, pvtInfo.npu_temp, pvtInfo.fan_speed);
+        snprintf(pcWriteBuffer, xWriteBufferLen,"cpu_temp(Celsius):%d.%d  npu_temp(Celsius):%d.%d  fan_speed(rpm):%d\n",
+            pvtInfo.cpu_temp/1000, pvtInfo.cpu_temp%1000,
+            pvtInfo.npu_temp/1000, pvtInfo.npu_temp%1000,
+            pvtInfo.fan_speed);
     }
 
     return pdFALSE;
@@ -1002,7 +1004,7 @@ static BaseType_t prvCommandPwrDissipationGet(char *pcWriteBuffer, size_t xWrite
     if (pwrStaus == SOM_POWER_ON) {
         get_board_power(&millivolt, &milliCur, &microWatt);
     }
-    snprintf(pcWriteBuffer, xWriteBufferLen,"consumption:%ld.%3ld(W)  voltage:%ld.%3ld(V)  current:%ld.%3ld(A)\n",
+    snprintf(pcWriteBuffer, xWriteBufferLen,"consumption:%ld.%3.3ld(W)  voltage:%ld.%03ld(V)  current:%ld.%03ld(A)\n",
         microWatt / 1000000, microWatt % 1000000, millivolt / 1000, millivolt % 1000, milliCur / 1000, milliCur % 1000);
 
     return pdFALSE;
diff --git a/Core/Src/hf_protocol_process.c b/Core/Src/hf_protocol_process.c
index edbbc52..561b4cf 100644
--- a/Core/Src/hf_protocol_process.c
+++ b/Core/Src/hf_protocol_process.c
@@ -628,7 +628,7 @@ void deamon_keeplive_task(void *argument)
 		if (old_status != get_som_daemon_state()) {
 			es_get_rtc_date(&date);
 			es_get_rtc_time(&time);
-			printf("SOM Daemon status change to %s at %d-%d-%d %02d:%02d:%02d!\n",
+			printf("SOM Daemon status change to %s at %d-%02d-%02d %02d:%02d:%02d!\n",
 				get_som_daemon_state() == SOM_DAEMON_ON ? "on" : "off",
 				date.Year, date.Month, date.Date, time.Hours, time.Minutes, time.Seconds);
 		}
-- 
2.25.1

